### PR TITLE
CORE-6308 Add Local Kubernetes deployment

### PR DESCRIPTION
**Pull request by davidcurrie**
Content predominantly sourced from https://github.com/corda/corda-runtime-os/wiki/Local-development-with-Kubernetes with updates to reflect changes in Docker Desktop and to remove content that is only applicable to Corda development.

Further updates will probably be needed to link this together with other content (e.g. for subsequent actions to upload a CPI, execute flows, ...) when it is available.

### CODE REVIEW

**Review by yarivus — COMMENTED** [review]
@davidcurrie / @colin-hillis / @nadinequinr3 
Should we be seeing these :
image

**Comment by davidcurrie** [comment]
> @davidcurrie / @colin-hillis / @nadinequinr3 Should we be seeing these : image

I was just following the example in https://r3-cev.atlassian.net/wiki/spaces/CB/pages/3767173243/Style+guide+for+technical+writing#Code-Blocks. I believe this is Hugo specific so won't show up in a GitHub preview. Let me know if it should be removed though.

**Comment by nadinequinr3** [comment]
Yes, the tabs are formatted by Hugo so wouldn't expect them to display correctly here. However, I have a feeling Hugo won't like the additional line breaks so will investigate that.

**Comment by davidcurrie** [comment]
> Yes, the tabs are formatted by Hugo so wouldn't expect them to display correctly here. However, I have a feeling Hugo won't like the additional line breaks so will investigate that.

That would be amusing as I only added them because the Markdown linter in my IDE insists that code blocks should be surrounded by empty lines!

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 1]
```diff
@@ -8,3 +8,221 @@ menu:
...
+## Build the Corda Docker images
+
+The Corda Docker images need to be built from source.
+
+1. If you’re using minikube, configure your shell to use the Docker daemon inside minikube so that built images are available directly to the cluster:
+
+    {{< tabs name="Setting Docker environment for minikube">}}
+    {{% tab name="Bash"%}}
+
+    ```bash
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:127]:
```suggestion
```bash
```

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 2]
```diff
@@ -8,3 +8,221 @@ menu:
...
+    eval $(minikube docker-env)
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:128]:
```suggestion
eval $(minikube docker-env)
```

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 3]
```diff
@@ -8,3 +8,221 @@ menu:
...
+    ```
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:129]:
```suggestion
```
```

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 4]
```diff
@@ -8,3 +8,221 @@ menu:
...
+
+    {{% /tab %}}
+
+    {{% tab name="PowerShell" %}}
+
+    ```pwsh
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:135]:
```suggestion
```pwsh
```

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 5]
```diff
@@ -8,3 +8,221 @@ menu:
...
+    minikube docker-env --shell=powershell | Invoke-Expression
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:136]:
```suggestion
minikube docker-env --shell=powershell | Invoke-Expression
```

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 6]
```diff
@@ -8,3 +8,221 @@ menu:
...
+    ```
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:137]:
```suggestion
```
```

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 7]
```diff
@@ -8,3 +8,221 @@ menu:
...
+    If you are using Docker Desktop, you can also switch context via the Kubernetes sub-menu.
+
+2. Create a namespace to contain your Corda deployment.
+    For example, to create a namespace called `corda` run the command:
+
+    ```sh
+    kubectl create namespace corda
+    ```
+
+**Note:** The commands that follow all assume that you are using a namespace called `corda`.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:86]:
```suggestion
{{< note >}}
The commands that follow all assume that you are using a namespace called `corda`.
```

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 8]
```diff
@@ -8,3 +8,221 @@ menu:
...
+Modify the `-n corda` option on each `kubectl` command if you use a different namespace.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:87]:
```suggestion
Modify the `-n corda` option on each `kubectl` command if you use a different namespace.
{{< /note >}}
```

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 9]
```diff
@@ -8,3 +8,221 @@ menu:
...
+    helm dependency build charts/corda-dev
+    ```
+
+3. Install the Helm chart:
+
+    ```sh
+    helm install prereqs -n corda charts/corda-dev --render-subchart-notes --timeout 10m --wait
+    ```
+
+    The `--wait` option ensures all of the pods are ready before returning and the `--render-subchart-notes` option gives you a brief overview of the connection details.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:114]:
```suggestion
    The `--wait` option ensures all of the pods are ready before returning. The `--render-subchart-notes` option gives you a brief overview of the connection details.
```

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 10]
```diff
@@ -8,3 +8,221 @@ menu:
...
+    The timeout is set to 10 minutes to allow time to pull the images from Docker Hub.
+    The process should take significantly less time than this on subsequent installs.
+
+## Build the Corda Docker images
+
+The Corda Docker images need to be built from source.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:120]:
```suggestion
The Corda Docker images must be built from source.
```

File: content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md [hunk 11]
```diff
@@ -8,3 +8,221 @@ menu:
...
+These values configure the chart to use the images you just built and specify the location of the Kafka and PostgreSQL instances created by the `corda-dev` Helm chart.
+They also set the initial admin user password to `admin`.
+
+1. Install the chart as follows by running from the root of the `corda-runtime-os` repository:
+
+   ```sh
+   helm install corda -n corda charts/corda --values values.yaml --wait
+   ```
+
+When the commmand completes, the RPC endpoint should ready to access.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-dev-preview-2/deploying/local-deployment.md:182]:
```suggestion
When the commmand completes, the RPC endpoint should be ready to access.
```

**Comment by nadinequinr3** [comment]
I think this piece will fit better under the "tutorials" section. I'm happy to move it after it's merged to the branch but I think then we still need a brief overview type piece for this section. Why you would deploy this way? The benefits/limitations etc

Just FYI 
image
I did some testing locally and the issue is with the indenting within the tabs shortcode.

**Review by nadinequinr3 — CHANGES_REQUESTED** [review]

**Comment by davidcurrie** [comment]
We'll see what it does with the numbering. I'd rather have the code indented correctly under  the numbering than have the tabs if it comes to it.

Happy to see this move under tutorials. I suspect you may end up dropping the whole `deploying` section for DP2.

It will be easier to add the introduction once the other material is in place as I can then cross link to what they probably should be doing instead. (The approach documented is a bit of a stop-gap until we have something better in the beta.)

**Comment by nadinequinr3** [comment]
> We'll see what it does with the numbering. I'd rather have the code indented correctly under the numbering than have the tabs if it comes to it.
> 
Sorry, once the {{ tabs }} part is indented the whole piece will be:

image

**Comment by davidcurrie** [comment]
Oh - great. That's perfect.

**Comment by davidcurrie** [comment]
Has the `C5-DP2` branch got rewritten at some point? I seem to have got a stack of commits on my PR branch that no longer appear on that branch and now lead to a pile of conflicts. Should I just close this and start again?

**Comment by nadinequinr3** [comment]
There would have been a lot of commits to those files listed (except local-deployment) over the last few days and CP5-DP2 was also rebased with main but I'm not sure why we're seeing conflicts. Your changes were all in that one file, right?

**Comment by nadinequinr3** [comment]
If it's ok with you, it might be cleanest/safest/fastest for me to just copy the contents with your latest changes to the branch directly? I want to add it to the file under "Tutorials" anyway..

**Comment by davidcurrie** [comment]
> If it's ok with you, it might be cleanest/safest/fastest for me to just copy the contents with your latest changes to the branch directly? I want to add it to the file under "Tutorials" anyway..

It'll be the rebase that's the problem. Sure - I'm fine with you just copying the contents and, yes, all my changes are in the one file.

**Comment by nadinequinr3** [comment]
You can view a preview here: https://nadine.preview.docs.r3.com/en/platform/corda/5.0-dev-preview-2/deploying/overview.html

**Pull request closed without merging**